smartcosmos-framework-monitoring/src/main/java/net/smartcosmos/metrics/MetricsConfiguration.java

```diff
@@ -0,0 +1,111 @@
+package net.smartcosmos.metrics;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import io.prometheus.client.exporter.MetricsServlet;
+import io.prometheus.client.spring.boot.SpringBootMetricsCollector;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.actuate.autoconfigure.ManagementWebSecurityAutoConfiguration;
+import org.springframework.boot.actuate.endpoint.MetricsEndpoint;
+import org.springframework.boot.actuate.endpoint.PublicMetrics;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.ConditionOutcome;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.boot.autoconfigure.condition.SpringBootCondition;
+import org.springframework.boot.autoconfigure.web.EmbeddedServletContainerAutoConfiguration;
+import org.springframework.boot.bind.RelaxedPropertyResolver;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.annotation.AnnotationAwareOrderComparator;
+import org.springframework.core.env.Environment;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+import org.springframework.web.servlet.mvc.ServletWrappingController;
+
+@Configuration
+@ConditionalOnWebApplication
+@ConditionalOnClass({ MetricsServlet.class, ServletWrappingController.class })
+@Conditional(MetricsConfiguration.PrometheusCondition.class)
+@AutoConfigureBefore(ManagementWebSecurityAutoConfiguration.class)
```

Comment:
@sdomula  was. 😉 

I basically moved his code here and added the annotation and configuration.